src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,432 @@
+package org.broadinstitute.hellbender.cmdline.GATKPlugin;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineException;
+import org.broadinstitute.barclay.argparser.CommandLinePluginDescriptor;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.walkers.annotator.Annotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.reflections.ReflectionUtils;
+
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A plugin descriptor for managing the dynamic discovery of both @{InfoFieldAnnotation} and @{GenotypeAnnotation} objects
+ * within the @{org.broadinstitute.hellbender.tools.walkers.annotator} package which handles integrating annotation specific
+ * arguments from the command line with tool specified defaults.
+ *
+ * Unlike @{GATKReadFilterPluginDescriptor} annotation order is not important and thus argument order is not guarinteed to be
+ * preserved in all cases, especially when group annotations are involved.
+ */
+public class GATKAnnotationPluginDescriptor  extends CommandLinePluginDescriptor<Annotation> {
+    private static final String pluginPackageName = "org.broadinstitute.hellbender.tools.walkers.annotator";
```

Comment:
Candidate for configuration for downstream...maybe a TODO to mark that will be nice!